homework/tests/example.spec.js

```diff
@@ -1,31 +1,4015 @@
 // @ts-check
 const { test, expect } = require('@playwright/test');
 
+test.beforeEach(async ({ page }) => {
+  await page.goto('https://testsheepnz.github.io/BasicCalculator');
+});
+
 const data = [
   'Prototype',
-  '1',
-  '2',
-  '3',
-  '4',
-  '5',
-  '6',
-  '7',
-  '8',
-  '9'
+  //'1',
+  //'2',
+  //'3',
+  //'4',
+  //'5',
+  //'6',
+  //'7',
+  //'8',
+  //'9'
 ]
 
```

Comment:
All the selectors in the code could described as constants if they do repeat. For example:

const firstNumberInputField = '#number1Field';

So then you can use it in code like if (await page.locator(firstNumberInputField ).isVisible())

Image situation if you have 2000 tests where selector is written in a way like you did, and it changes to something like '#numberXField' - now you would need to refactor code and change it in every single place instead of easily updating the selector constant.